contracts/V1/StateChannelDiamondProxy/DisputeManagerFacet.sol

```diff
@@ -227,22 +273,96 @@ contract DisputeManagerFacet is StateChannelCommon {
         }
     }
 
+    function challengeDisputeReduction(
+        Dispute[] memory disputes,
+        uint256[] disputeCreationTimestamps,
+        uint256 disputeWindowExpirationTimestamp,
+        StateSnapshot memory stateSnapshot,
+        bytes memory encodedStateMachineState,
+        JoinChannelBlock[] memory joinChannelBlocks
+    ) public {
+        require(disputes.length > 0, ErrorNoDisputesProvided());
+        bytes32 channelId = disputes[0].channelId;
+        require(_canParticipateInDisputes(channelId, msg.sender), ErrorCantParticipateInDispute());
+        DisputeData storage disputeData = disputeData[channelId];
+        DisputeWindow storage disputeWindow = disputeData.disputeWindowMap[disputes[0].genesisSnapshotDataHash];
+
+        //rquire all disputes are part of commitment
+
+        require(
+            areDisputesCommitted(disputeWindow, disputes, disputeCreationTimestamps),
+            ErrorDisputeCommitmentNotAvailable()
+        );
+        //require reduce challenge period is not expired - this also ashures it's commited
+        require(!_isReduceChallengePeriodExpired(disputeWindow), ErrorDisputeChallengePeriodExpired());
+
+        ReduceOutput memory reducedOutput = reduce(disputes, disputeWindowExpirationTimestamp);
+
+        SnapshotData memory snapshotData =
+            reduceOutputToSnapshotData(reducedOutput, stateSnapshot, encodedStateMachineState, joinChannelBlocks);
+
+        bytes32 winingForkId = keccak256(abi.encode(snapshotData));
+
+        if (winingForkId != disputeWindow.reducedResult.reducedForkId) {
+            addOnChainSlashedParticipants(channelId, disputeWindow.reducedResult.reducer);
+            disputeWindow.reducedResult.reducedForkId = bytes32(0); // unset
+            _commitToDisputeReducedResult(disputeWindow, winingForkId, block.timestamp - getChallengeTime() - 1);
+        } else {
```

Comment:
im guessing the `-1` is because somehwere the expiration check uses `<`
could use `<=` instead and remove the `-1`

just a guess